### PR TITLE
More exact correspondence of getmetatatable to vanilla Lua

### PR DESCRIPTION
**Pull request by foul11**

### CODE REVIEW

File: lua/starfall/libs_sh/builtins.lua [hunk 1]
```diff
@@ -195,11 +195,13 @@ builtins_library.isbool = isbool
 -- @return boolean If the object is a function or not
 builtins_library.isfunction = isfunction
 
---- Returns the metatable of an object. Doesn't work on most internal metatables
+--- Returns the metatable of an object.
+-- Doesn't work on most internal metatables.
+-- For any types other than table, nil will be returned.
```
Comment (Vurv78) [inline lua/starfall/libs_sh/builtins.lua:200]: Change return type to `table?` then

Comment (Vurv78) [reply lua/starfall/libs_sh/builtins.lua:200]: Although I think getmetatable can return anything depending on the ``__metatable``, so either return `any?` or `table?`

**Review by Vurv78 — CHANGES_REQUESTED** [review]

**Review by foul11 — COMMENTED** [review]
fix docs

File: lua/starfall/libs_sh/builtins.lua (hunk 1 above)
Comment (foul11) [reply lua/starfall/libs_sh/builtins.lua:200]: I think you shouldn’t take into account that there is usually a __metatable property, ideally the getmetatable function should return a table, __metatable in this case acts as an extension of the basic capabilities, as if __add returned a string

**Review by Vurv78 — APPROVED** [review]

**Pull request merged**